dataclients/kubernetes/routegroup.go

```diff
@@ -513,49 +518,120 @@ func transformRouteGroup(ctx *routeGroupContext) ([]*eskip.Route, error) {
 	var rs []*eskip.Route
 
-	hostRoutes := make(map[string][]*eskip.Route)
 	for _, rg := range s.routeGroups {
+		var internalHosts []string
+		var externalHosts []string
+
 		hosts := rg.Spec.UniqueHosts()
-		ctx := &routeGroupContext{
-			clusterState:          s,
-			defaultFilters:        df,
-			routeGroup:            rg,
-			hosts:                 hosts,
-			hostRx:                createHostRx(hosts...),
-			hostRoutes:            make(map[string][]*eskip.Route),
-			hasEastWestHost:       hasEastWestHost(r.options.KubernetesEastWestDomain, hosts),
-			eastWestEnabled:       r.options.KubernetesEnableEastWest,
-			eastWestDomain:        r.options.KubernetesEastWestDomain,
-			provideHTTPSRedirect:  r.options.ProvideHTTPSRedirect,
-			httpsRedirectCode:     r.options.HTTPSRedirectCode,
-			backendsByName:        mapBackends(rg.Spec.Backends),
-			backendNameTracingTag: r.options.BackendNameTracingTag,
+		if len(r.options.KubernetesEastWestRangeDomains) == 0 {
+			externalHosts = hosts
+		} else {
+			internalHosts, externalHosts = splitHosts(hosts, r.options.KubernetesEastWestRangeDomains)
 		}
 
-		ri, err := transformRouteGroup(ctx)
-		if err != nil {
-			log.Errorf(
-				"[routegroup] error transforming %s/%s: %v.",
-				namespaceString(rg.Metadata.Namespace),
-				rg.Metadata.Name,
-				err,
-			)
+		backends := mapBackends(rg.Spec.Backends)
+
+		// If there's no host at all, or if there's any external hosts
+		// create it.
+		if len(externalHosts) != 0 || len(hosts) == 0 {
+			ctx := &routeGroupContext{
+				clusterState:          s,
+				defaultFilters:        df,
+				routeGroup:            rg,
+				hosts:                 externalHosts,
+				hostRx:                createHostRx(externalHosts...),
+				hostRoutes:            make(map[string][]*eskip.Route),
+				hasEastWestHost:       hasEastWestHost(r.options.KubernetesEastWestDomain, externalHosts),
+				eastWestEnabled:       r.options.KubernetesEnableEastWest,
+				eastWestDomain:        r.options.KubernetesEastWestDomain,
+				provideHTTPSRedirect:  r.options.ProvideHTTPSRedirect,
+				httpsRedirectCode:     r.options.HTTPSRedirectCode,
+				backendsByName:        backends,
+				backendNameTracingTag: r.options.BackendNameTracingTag,
+				internal:              false,
+			}
 
-			continue
+			ri, err := transformRouteGroup(ctx)
+			if err != nil {
+				log.Errorf(
+					"[routegroup] error transforming external hosts for %s/%s: %v.",
+					namespaceString(rg.Metadata.Namespace),
+					rg.Metadata.Name,
+					err,
+				)
+
+				continue
+			}
+
+			catchAll := hostCatchAllRoutes(ctx.hostRoutes, func(host string) string {
+				// "catchall" won't conflict with any HTTP method
+				return rgRouteID("", toSymbol(host), "catchall", 0, 0, false)
+			})
+			ri = append(ri, catchAll...)
+
+			rs = append(rs, ri...)
 		}
 
-		rs = append(rs, ri...)
-		mergeHostRoutes(hostRoutes, ctx.hostRoutes)
-	}
+		// Internal hosts
+		for _, host := range internalHosts {
```

Comment:
What is the reason here to loop over internal domains where for external domains we just combine them
```go
hostRx:                createHostRx(externalHosts...),
```
@jonathanbeber 
Why not to do the same here, i.e. get rid of the loop and use single `internalCtx`
```go
hostRx:                createHostRx(internalHosts...),
```

Reply:
You're right, that was due to an [initial implementation](https://github.com/zalando/skipper/commit/b72f8d47cbda834395913171da2643dd466d0631#diff-3a9a80b20ebd6d5f62df3e5a8a080e2e022bb1ab9ec7a2089021f236f0f85b14R630) where we could define multiple IPs for different domains. The idea is that you could have `domain.internal` to the CIDR `10.0.0/8`, and `domain2.internal`  to the CIDR `192,168,0.0/24` for example. 

I will fix it.

Reply:
Thank you

Reply:
#1758 should handle that.